File: src/main/resources/output/IFC4X3_python/Entities/IfcRepresentationItem/IfcGeometricRepresentationItem/IfcSegment/IfcCompositeCurveSegment/IfcCompositeCurveSegment.py
class IfcCompositeCurveSegment(IfcSegment):
    def __init__(self):
        self.SameSense: IfcBoolean = None
        self.ParentCurve: IfcCurve = None

    # === EXTENDED BY ===
    # IfcReparametrisedCompositeCurveSegment

    # === WHERE CLAUSES ===
    # ParentIsBoundedCurve : ('IFC4X3_DEV_73740fe4.IFCBOUNDEDCURVE' IN TYPEOF(ParentCurve))

File: output/ifc4x3_dev_73740fe4/entity/IfcCompositeCurveSegment.java
package com.topoplanet.ifc4x3_dev_73740fe4.entity;
import com.topoplanet.test.type.*;


public class IfcCompositeCurveSegment extends IfcSegment {

    public IfcBoolean SameSense;
    public IfcCurve ParentCurve;
    // INVERSE attributes:
    // ParentIsBoundedCurve : ('IFC4X3_DEV_73740fe4.IFCBOUNDEDCURVE' IN TYPEOF(ParentCurve));

    // SUPERTYPE OF:
    // - IfcReparametrisedCompositeCurveSegment SUBTYPE OF IfcSegment

    // WHERE constraints:
    // ParentIsBoundedCurve : ('IFC4X3_DEV_73740fe4.IFCBOUNDEDCURVE' IN TYPEOF(ParentCurve));
}
